lib/process-services-cloud/src/lib/form/services/form-cloud.service.ts

```diff
@@ -0,0 +1,187 @@
+        );
+    }
+
+    saveTaskForm(appName: string, taskId: string, formId: string, formValues: FormValues) {
+        const apiUrl = this.buildSaveFormUrl(appName, formId);
+        const saveFormRepresentation = <SaveFormRepresentation> { values: formValues, taskId: taskId };
+        return from(this.apiService
+            .getInstance()
+            .oauth2Auth.callCustomApi(apiUrl, 'POST',
+                null, null, null,
+                null, saveFormRepresentation,
+                this.contentTypes, this.accepts,
+                this.returnType, null, null)
+        ).pipe(
+            map((res: any) => {
+                return new TaskDetailsCloudModel(res.entry);
+            }),
+            catchError((err) => this.handleError(err))
+        );
+    }
+
+    completeTaskForm(appName: string, taskId: string, formId: string, formValues: FormValues, outcome: string) {
+        const apiUrl = this.buildSubmitFormUrl(appName, formId);
+        let completeFormRepresentation: any = <CompleteFormRepresentation> { values: formValues, taskId: taskId };
+        if (outcome) {
+            completeFormRepresentation.outcome = outcome;
+        }
+
+        return from(this.apiService
+            .getInstance()
+            .oauth2Auth.callCustomApi(apiUrl, 'POST',
+                null, null, null,
+                null, completeFormRepresentation,
+                this.contentTypes, this.accepts,
+                this.returnType, null, null)
+        ).pipe(
+            map((res: any) => {
+                return new TaskDetailsCloudModel(res.entry);
+            }),
+            catchError((err) => this.handleError(err))
+        );
+    }
+
+    getTask(appName: string, taskId: string): Observable<TaskDetailsCloudModel> {
+        const apiUrl = this.buildGetTaskUrl(appName, taskId);
+        return from(this.apiService
+            .getInstance()
+            .oauth2Auth.callCustomApi(apiUrl, 'GET',
+                null, null, null,
+                null, null,
+                this.contentTypes, this.accepts,
+                this.returnType, null, null)
+        ).pipe(
+            map((res: any) => {
+                return new TaskDetailsCloudModel(res.entry);
+            }),
+            catchError((err) => this.handleError(err))
+        );
+    }
+
+    getTaskVariables(appName: string, taskId: string): Observable<any[]> {
+        const apiUrl = this.buildGetTaskVariablesUrl(appName, taskId);
+        return from(this.apiService
+            .getInstance()
+            .oauth2Auth.callCustomApi(apiUrl, 'GET',
+                null, null, null,
+                null, null,
+                this.contentTypes, this.accepts,
+                this.returnType, null, null)
+        ).pipe(
+            map((res: any) => {
+                return res.content;
+            }),
+            catchError((err) => this.handleError(err))
+        );
+    }
+
+    getForm(appName: string, taskId: string): Observable<any> {
```

Comment:
why any?

Reply:
This service returns the form json as it is. If need to use form-cloud-model then we'll have to parse it. There is one more method available to do that.